gammapy/cube/models.py

```diff
@@ -395,3 +426,77 @@ def evaluate(self):
         tilt_factor = np.power((self.energy_center / reference).to(""), -tilt)
         back_values = norm * self.map.data * tilt_factor.value
         return self.map.copy(data=back_values)
+
+    @classmethod
+    def from_skymodel(cls, skymodel, exposure, edisp=None, psf=None, **kwargs):
```

Comment:
This is definitely something we have to re-discuss. Currently we support a constant or galactic diffuse background component via `SkyDiffuseCube` or `SkyDiffuseConstant`. Unfortunately this is super inefficient with the `MapEvaluator`, because the model is re-evaluated (i.e interpolated,  PSF-convolved and energy migrated) in every iteration of the fit. The effect here is not minor, it slows down the fitting by a factor of ~50 depending on the image size and becomes the bottleneck of the model evalutaion. Currently we have the `fermi_lat.ipynb` and `analysis_3d.ipynb` notebook which are affected by this. The general question is probably more, how to handle galactic diffuse emission. Is it handled as a background or source component? I guess it depends on you analysis interest. In the use cases listed above its probably part of the background...

The solution here is of course to precompute the models on the analysis grid and then just scale and tilt them. Which is exactly, what the `BackgroundModel` class does. So this is the idea behind the `.from_skymodel()` method.

I guess the better solution would be to have an internal caching of the `SkyDiffuseCube` or `SkyDiffuseConstant` models, but then we need some mechanism to couple it with the spectral parameters. I'm happy for any better idea / suggestion how to handle this...

Reply:
A compromise for now could be to just use the `MapEvaluator` directly in the notebooks and pre-computed the diffuse background model and adding a comment that it's needed because of performance. Then I would delete the `.from_skymodel()` method again.

Reply:
OK I see the point now. 

Technically, changing the spectral index on a `SkyModel` and applying a tilt on the `BackgroundModel` should not be identical since the `edisp` should redistribute in some non-trivial manner. 

That said I have no clear idea on how to tell the `MapEvaluator` to cache the result of the evaluation of `apply_psf` when the model is a `SkyDiffuseCube` or `SkyDiffuseMap`.

I guess the `.from_skymodel()` method is OK for now, as there might be other uses cases for it.